lnrpc/invoicesrpc/addinvoice.go

```diff
@@ -847,6 +848,378 @@ func PopulateHopHints(cfg *SelectHopHintsCfg, amtMSat lnwire.MilliSatoshi,
 	return hopHints, nil
 }
 
+// buildBlindedPathCfg defines the various resources required to build a blinded
+// payment path to this node.
+type buildBlindedPathCfg struct {
+	// findRoutes returns a set of routes to us that can be used for the
+	// construction of blinded paths.
+	findRoutes func(value lnwire.MilliSatoshi) ([]*route.Route, error)
+
+	// fetchChannelEdgesByID attempts to look up the two directed edges for
+	// the channel identified by the channel ID.
+	fetchChannelEdgesByID func(chanID uint64) (*models.ChannelEdgeInfo,
+		*models.ChannelEdgePolicy, *models.ChannelEdgePolicy, error)
+
+	// bestHeight can be used to fetch the best block height that this node
+	// is aware of.
+	bestHeight func() (uint32, error)
+
+	// addPolicyBuffer is a function that can be used to alter the policy
+	// values of the given channel edge. The main reason for doing this is
+	// to add a safety buffer so that if the node makes small policy changes
+	// during the lifetime of the blinded path, then the path remains valid.
+	addPolicyBuffer func(
+		policy *models.ChannelEdgePolicy) (*bufferedChanPolicy, error)
+
+	// pathID is the secret data to embed in the blinded path data that we
+	// will receive back as the recipient.
+	pathID []byte
+
+	// valueMsat is the payment amount in milli-satoshis that must be
+	// routed. This will be used for selecting appropriate routes to use for
+	// the blinded path.
+	valueMsat lnwire.MilliSatoshi
+
+	// minFinalCLTVExpiryDelta is the minimum CLTV delta that the recipient
+	// requires for the final hop of the payment.
+	minFinalCLTVExpiryDelta uint32
+
+	// blocksUntilExpiry is the number of blocks that this blinded path
```

Comment:
> why would we want to change minFinalCtlvdelta to 144?

I am talking explicitly about this check here:

```
if cfg.MinFinalCLTVExpiryDelta >= cfg.BlocksUntilExpiry {
		return nil, fmt.Errorf("blinded path CLTV expiry delta (%d) "+
			"must be greater than the minimum final CLTV expiry "+
			"delta (%d)", cfg.BlocksUntilExpiry,
			cfg.MinFinalCLTVExpiryDelta)
	}
```

because now in the blinded case we as a receiver define the delta, we can be more explicit and set a limit exactly where the invoice really expires not our buffer, otherwise we are kind of invalidating our own invoice in the first place while creating it, so I would just use the real expiry here instead 

and maybe do the check in the addinvoice func:
```
	finalCLTVDelta += uint32(routing.BlockPadding)

        if finalCLTVDelta > invoiceExpiry {
               return error
       }
```

not sure if I am misunderstanding something here tho 😅

Reply:
> can you elaborate on this part pls?


So the user can actually provide an invoice with a manual expiry:

```
case invoice.Expiry > 0:

		// We'll ensure that the specified expiry is restricted to sane
		// number of seconds. As a result, we'll reject an invoice with
		// an expiry greater than 1 year.
		maxExpiry := time.Hour * 24 * 365
		expSeconds := invoice.Expiry

		if float64(expSeconds) > maxExpiry.Seconds() {
			return nil, nil, fmt.Errorf("expiry of %v seconds "+
				"greater than max expiry of %v seconds",
				float64(expSeconds), maxExpiry.Seconds())
		}

		expiry = time.Duration(invoice.Expiry) * time.Second

```

so basically the `BlocksUntilExpiry` can be very high, so we also need to make sure our finalCltvDelta is not greater than 2016 blocks, otherwise most of the nodes will reject the payment in the first place because of the default `max_cltv_delta` when trying to send a payment to that blinded path ?

Reply:
so my main point is, that we should probably execute more sane value checks for the finalCltvDelta when creating a blinded path. Tho this is a non-blocking comment and it only maybe improves robustness in a sense.